stm32-modules/heater-shaker/simulator/comm_thread.cpp

```diff
@@ -39,16 +40,37 @@ auto comm_thread::build()
 }
 
 auto comm_thread::handle_input(tasks::Tasks<SimulatorMessageQueue>& tasks)
-    -> void {
-    auto linebuf = std::make_shared<std::string>(1024, 'c');
-    while (true) {
-        if (!std::cin.getline(linebuf->data(), linebuf->size() - 1, '\n')) {
-            return;
+-> void {
+    boost::asio::io_service io_context;
+
+    boost::asio::ip::tcp::socket mysocket(io_context);
+    boost::asio::ip::tcp::endpoint endpoint(
+            boost::asio::ip::address::from_string("127.0.0.1"), 9999);
+    boost::system::error_code ec;
+    mysocket.connect(endpoint, ec);
+    if (ec) {
+
+    }
+
+    char pBuff[30];
+    boost::asio::mutable_buffer buff(pBuff, sizeof(pBuff));
+    std::string tot;
+
+    std::size_t l = mysocket.read_some(buff);
```

Comment:
this probably wants to be inside the loop. the way this is typically structured in c++ would be
```c++
std::size_t bytes_read = 0;
do {
   l = mysocket.read_some(buff);
   ...
} while (l > 0);
```
`do...while` is a loop structure that works the same as `while` but checks the condition at the end of the loop body rather than the beginning. It would require making sure the loop is resilient to a zero-byte read, however.

Reply:
I left this as a regular `while` loop because having it as a `do while` was causing it to only read a single G-Code and I couldn't figure out why